recipes/cuda-nvcc/meta.yaml

```diff
@@ -0,0 +1,56 @@
+{% set name = "cuda-nvcc" %}
+{% set version = "12.0.76" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/cuda_nvcc/{{ platform }}/cuda_nvcc-{{ platform }}-{{ version }}-archive.{{ extension }}
+
+  sha256: 160ca9f8828f53daa4b2066c5361aacb45fa2575885f70c223cda1d11df53d6f  # [linux64]
+  sha256: dc5bd7eac57b32a9645c0c23c6a1ba8a9533e846eeec31414e1aa4230afee2f8  # [ppc64le]
+  sha256: 85c8c643ea2cfa398e1b5ffcd597f5bb3b738526c9e3fb8e39dd909c55345f36  # [aarch64]
+  sha256: f67406f44263f0e4a110a8cf0200bff1d3d0e32917db8a7c9e417a87eb7ca372  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+  run_exports:
+    strong:
+      - __glibc >=2.17        # [linux]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  host:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+  run:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+
+test:
+  commands:
+    - nvcc --version
```

Comment:
It would maybe be nice to have a small test file that gets compiled and runs (even if it doesn't have any device code) just to make sure nothing is breaking in the usage path.

We can absolutely do this in follow up PRs though.

Reply:
Was thinking about that too. We can borrow [this test]( https://github.com/conda-forge/nvcc-feedstock/blob/9c8e43393d9c798c52a9d8e56785920ef767cabb/recipe/meta.yaml#L68-L74 )

Reply:
Proposed something in PR ( https://github.com/adibbley/staged-recipes/pull/1 )

Reply:
As part of creating this test, we realized we need to package `cuda-cudart`. Alex put together a draft PR with that recipe ( https://github.com/conda-forge/staged-recipes/pull/21723 )

Reply:
A test has been added. Depends on `cuda-cudart`, which is being packaged separately